bundles/org.openhab.binding.carnet/src/main/java/org/openhab/carnet/internal/services/CarNetVehicleServiceStatus.java

```diff
@@ -0,0 +1,264 @@
+                        Channel channel = thingHandler.getThing()
+                                .getChannel(definition.groupName + "#" + definition.channelName);
+                        if (channel != null) {
+                            logger.debug("Updading channel {} with value {}", channel.getUID(), getString(field.value));
+                            switch (definition.itemType) {
+                                case ITEMT_SWITCH:
+                                    updateSwitchChannel(channel, definition, field);
+                                    break;
+                                case ITEMT_STRING:
+                                    thingHandler.updateChannel(channel, new StringType(getString(field.value)));
+                                    break;
+                                case ITEMT_NUMBER:
+                                case ITEMT_PERCENT:
+                                default:
+                                    updateNumberChannel(channel, definition, field);
+                            }
+                        } else {
+                            logger.debug("Channel {}#{} not found", definition.groupName, definition.channelName);
+                        }
+
+                        if ((field.value != null) && !field.value.isEmpty()) {
+                            vehicleLocked &= checkLocked(field, definition);
+                            maintenanceRequired |= checkMaintenance(field, definition);
+                            tiresOk &= checkTires(field, definition);
+                            windowsClosed &= checkWindows(field, definition);
+                        }
+                    }
+                } else {
+                    logger.debug("{}: Unknown data field  {}.{}, value={} {}", thingId, data.id, field.id, field.value,
+                            field.unit);
+                }
+            }
+        }
+
+        // Update aggregated status
+        thingHandler.updateChannel(CHANNEL_GROUP_GENERAL, CHANNEL_GENERAL_LOCKED,
+                vehicleLocked ? OnOffType.ON : OnOffType.OFF); // Do we need this channel?
+        thingHandler.updateChannel(CHANNEL_GROUP_CONTROL, CHANNEL_CONTROL_LOCK,
+                vehicleLocked ? OnOffType.ON : OnOffType.OFF);
+        thingHandler.updateChannel(CHANNEL_GROUP_GENERAL, CHANNEL_GENERAL_MAINTREQ,
+                maintenanceRequired ? OnOffType.ON : OnOffType.OFF);
+        thingHandler.updateChannel(CHANNEL_GROUP_GENERAL, CHANNEL_GENERAL_TIRESOK,
+                tiresOk ? OnOffType.ON : OnOffType.OFF);
+        thingHandler.updateChannel(CHANNEL_GROUP_GENERAL, CHANNEL_GENERAL_WINCLOSED,
+                windowsClosed ? OnOffType.ON : OnOffType.OFF);
+
+        return true;
+    }
+
+    private boolean checkMaintenance(CNStatusField field, ChannelIdMapEntry definition) {
+        if (definition.symbolicName.contains("MAINT_ALARM") && field.value.equals(String.valueOf(1))) {
```

Comment:
What about this? Same for the others.
```suggestion
        if (definition.symbolicName.contains("MAINT_ALARM") && field.value.equals("1")) {
```

Reply:
changed